tests/ut/language/parser/test_submit_predicate.py

```
@@ -373,3 +373,232 @@ def main(
+    assert isinstance(dict(scope.attrs.items())["predicate"], ir.Gt)
+
+
+def test_scope_predicate_operand_producer_must_be_in_deps():
+    """Same contract as the call form: omitting deps= is caught."""
+    with pytest.raises(ParserSyntaxError, match="produced by the task"):
+        _scope_program("rc[0, 0] > 0", deps_src="")
+
+
+def test_scope_predicate_rejected_inside_cluster():
+    """A cluster-nested pl.spmd never becomes a Submit, so the predicate is lost."""
+    with pytest.raises(ParserSyntaxError, match="cannot be nested inside `pl.cluster"):
```

Ruff (0.15.21)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)